crypto/crypto_aes.c

```diff
@@ -154,12 +178,19 @@ hwaccel_init(void)
 
 #if defined(CPUSUPPORT_X86_AESNI)
 	CPUSUPPORT_VALIDATE(hwaccel, HW_X86_AESNI, cpusupport_x86_aesni(),
-	    hwtest(x86_aesni_oneshot));
+	    functest(x86_aesni_oneshot));
 #endif
 #if defined(CPUSUPPORT_ARM_AES)
 	CPUSUPPORT_VALIDATE(hwaccel, HW_ARM_AES, cpusupport_arm_aes(),
-	    hwtest(arm_aes_oneshot));
+	    functest(arm_aes_oneshot));
 #endif
+
+	/*
+	 * If we're here, we're not using any intrinsics.  Test OpenSSL; if
+	 * there's an error, print a warning, but continue.
+	 */
+	if (functest(openssl_oneshot))
+		warn0("OpenSSL gives incorrect AES values.");
```

Comment:
I would be inclined to add `abort()` here.  If we don't have any working AES routines, we can't do anything useful anyway.